src/rownode.cpp

```diff
@@ -72,12 +77,15 @@ RowNode* RowNode::createNode(QMap<QString, QVariant> bookItem, QMap<QString, QBy
             bookIcon = iconMap[faviconUrl];
         }
     }
-    auto temp = new RowNode({bookIcon, bookItem["title"],
+    std::weak_ptr<RowNode> weakRoot = rootNode;
+    auto rowNodePtr = std::make_shared<RowNode>(
+                                    RowNode({bookIcon, bookItem["title"],
```

Comment:
Don't use `make_shared` must directly init a std::shared_ptr with a pointer:
```cpp
auto rowNodePtr = std::shared_ptr<RowNode>(new RowNode([...]));
```

(Does like in the "best" example of the `shared_from_this` doc)